packages/cli/src/modules/breaking-changes/rules/v2/process-env-access.rule.ts

```diff
@@ -0,0 +1,81 @@
+import { Service } from '@n8n/di';
+
+import type { DetectionResult, BreakingChangeMetadata, CommonDetectionInput } from '../../types';
+import { BreakingChangeSeverity, BreakingChangeCategory, IssueLevel } from '../../types';
+import { AbstractBreakingChangeRule } from '../abstract-rule';
+
+@Service()
+export class ProcessEnvAccessRule extends AbstractBreakingChangeRule {
+	getMetadata(): BreakingChangeMetadata {
+		return {
+			id: 'process-env-access-v2',
+			version: 'v2',
+			title: 'Block process.env Access in Expressions',
+			description: 'Direct access to process.env is blocked by default for security',
+			category: BreakingChangeCategory.WORKFLOW,
+			severity: BreakingChangeSeverity.HIGH,
+		};
+	}
+
+	async detect({ workflows }: CommonDetectionInput): Promise<DetectionResult> {
+		const result = this.createEmptyResult();
+
+		try {
+			const processEnvPattern = /process\.env/;
+
+			for (const workflow of workflows) {
+				const affectedNodes: string[] = [];
+
+				workflow.nodes.forEach((node) => {
+					// Check in Code nodes
+					if (node.type === 'n8n-nodes-base.code') {
+						const code =
+							typeof node.parameters?.code === 'string' ? node.parameters.code : undefined;
+						if (code && processEnvPattern.test(code)) {
+							affectedNodes.push(node.name);
+						}
+					} else {
+						// Check in expressions
+						const nodeJson = JSON.stringify(node.parameters);
+						if (processEnvPattern.test(nodeJson) && !affectedNodes.includes(node.name)) {
+							affectedNodes.push(node.name);
+						}
+					}
+				});
+
+				if (affectedNodes.length > 0) {
+					result.affectedWorkflows.push({
+						id: workflow.id,
+						name: workflow.name,
+						active: workflow.active,
+						issues: [
+							{
+								title: 'process.env access detected',
+								description: `The following nodes contain process.env access: ${affectedNodes.join(', ')}. This will be blocked in v2.0.0.`,
```

Comment:
```suggestion
								description: `The following nodes contain process.env access: ${affectedNodes.join(', ')}. This will be blocked by default in v2.0.0.`,
```